### PR TITLE
ci: Add hive to pipeline

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -593,6 +596,50 @@ jobs:
           name: "Go mod tidy"
           command: make mod-tidy && git diff --exit-code
 
+  build-hive:
+    docker:
+      - image: cimg/go:1.18
```
Comment (trianglesphere) [inline .circleci/config.yml:601]: maybe separate, but switch to go1.19?

**Comment by trianglesphere** [comment]
an intermediate build step failed

File: .circleci/config.yml (hunk 1 above)
Comment (mslipper) [reply .circleci/config.yml:601]: Suce, I can do that. Hive builds on ancient versions of Go so it shouldn't be a big issue.

**Comment by mslipper** [comment]
Yeah, seeing I/O errors pulling Docker containers. Looking into it.

**Review by tynes — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -593,6 +596,40 @@ jobs:
...
+      - go/mod-download
+      - go/save-cache
+      - run: {command: "go build ."}
+      - run:
+          command: |
+            ./hive \
+            -sim=<<parameters.sim>> \
+            -sim.loglevel=5 \
+            -docker.pull=true \
+            -client=go-ethereum,op-geth_optimism-history,op-proposer_<<parameters.version>>,op-batcher_<<parameters.version>>,op-node_<<parameters.version>>,op-contracts_<<parameters.version>> |& tee /tmp/hive.log
```
Comment (protolambda) [inline .circleci/config.yml:623]: Is there some way to confirm it pulls the correct images for the hive client containers, i.e. if the images that we pushed as dependency are available soon enough on the docker repo to be used in the CI hive build?

**Review by protolambda — APPROVED** [review]
Question about cycle time between image build and hive usage, but LGTM otherwise.

File: .circleci/config.yml (hunk 2 above)
Comment (mslipper) [reply .circleci/config.yml:623]: Hive will fail if the passed-in image doesn't exist on the repo.

Comment (mslipper) [reply .circleci/config.yml:623]: I confirmed it's pulling the correct containers by cross referencing the params with the logs.

**Pull request merged**